src/api/app/models/project.rb

```diff
@@ -125,77 +120,364 @@ def staging_history
 
   validates :kind, inclusion: { in: TYPES }
 
-  def self.home?(name)
-    name.start_with?('home:')
-  end
+  class << self
+    def home?(name)
+      name.start_with?('home:')
+    end
 
-  def self.deleted?(project_name)
-    return false if find_by_name(project_name)
+    # NOTE: This has to cover project name validations in src/backend/BSVerify.pm (verify_projid)
+    def valid_name?(name)
+      return false unless name.is_a?(String)
+      return false if name == '0'
+      return false if name =~ /:[:\._]/
+      return false if name =~ /\A[:\._]/
+      return false if name.end_with?(':')
+      return true  if name =~ /\A[-+\w\.:]{1,200}\z/
```

Comment:
:thought_balloon: I would find super useful to have those patterns as constants, with names explaining what they match

Reply:
Ok, I remembered this is  not a refactoring but a reorganization. So feel free to discard my comment.